From 22d61cd0891d8304dfc1a70579cf154fd8e6644a Mon Sep 17 00:00:00 2001
From b33c3cb9651459de1f3d549677cbac67a017a295 Mon Sep 17 00:00:00 2001
From: Daniel Drake <dan@reactivated.net>
Date: Tue, 22 Jun 2010 18:20:23 -0500
Subject: [PATCH] Linux: fix log message that was being unconditionally sent to stderr (#44)

This message was put in place to aid libusb-compat-0.1 users, who
may run into the change that libusb-0.1 allowed you to usb_open() a
device that you don't have write access to, but libusb-1.0 does not.

As explained on the ticket this change is dangerous so it should go through
the usual logging mechanism. I also added a similar regular log message
to libusb-compat-0.1.

Resolves: #628356

diff -uNPrp libusb-1.0.8/libusb/os/linux_usbfs.c libusb-1.0.8.new/libusb/os/linux_usbfs.c
--- libusb-1.0.8/libusb/os/linux_usbfs.c	2010-04-22 23:14:46.000000000 +0200
+++ libusb-1.0.8.new/libusb/os/linux_usbfs.c	2010-09-14 16:22:31.314772148 +0200
@@ -1055,10 +1055,10 @@ static int op_open(struct libusb_device_
 	hpriv->fd = open(filename, O_RDWR);
 	if (hpriv->fd < 0) {
 		if (errno == EACCES) {
-			fprintf(stderr, "libusb couldn't open USB device %s: "
-				"Permission denied.\n"
-				"libusb requires write access to USB device nodes.\n",
-				filename);
+			usbi_err(HANDLE_CTX(handle), "libusb couldn't open USB device %s: "
+				"Permission denied.", filename);
+			usbi_err(HANDLE_CTX(handle),
+				"libusb requires write access to USB device nodes.");
 			return LIBUSB_ERROR_ACCESS;
 		} else if (errno == ENOENT) {
 			return LIBUSB_ERROR_NO_DEVICE;
